pkg/apiserver/logsearch/router.go

```diff
@@ -0,0 +1,353 @@
+			fmt.Errorf("cannot start, task group is %s", StateRunning))
+		return
+	}
+	s.scheduler.addTasks(components, searchLogReq)
+	err := s.scheduler.runTaskGroup(false)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+	httputil.Success(c)
+}
+
+// @Summary Download logs by task ID
+// @Description download logs by task id
+// @Produce application/zip
+// @Param id path string true "task id"
+// @Failure 400 {object} httputil.HTTPError
+// @Failure 500 {object} httputil.HTTPError
+// @Router /logs/tasks/{id}/download [get]
+func (s *Service) TaskDownload(c *gin.Context) {
+	taskID := c.Param("id")
+	task, err := s.queryTaskByID(taskID)
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	f, err := os.Open(task.SavedPath)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+	_, err = io.Copy(c.Writer, f)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+}
+
+func (s *Service) queryTaskByID(taskID string) (task TaskModel, err error) {
+	err = s.db.First(&task, "task_id = ?", taskID).Error
+	return
+}
+
+// @Summary Download logs by task IDs
+// @Description download logs by multiple task IDs
+// @Produce application/x-tar
+// @Param id query string false "task id"
+// @Failure 400 {object} httputil.HTTPError
+// @Failure 500 {object} httputil.HTTPError
+// @Router /logs/download [get]
+func (s *Service) MultipleTaskDownload(c *gin.Context) {
+	ids := c.QueryArray("id")
+	tasks := make([]*TaskModel, 0, len(ids))
+	for _, taskID := range ids {
+		task, err := s.queryTaskByID(taskID)
+		if err != nil {
+			httputil.NewError(c, http.StatusBadRequest, err)
+			return
+		}
+		tasks = append(tasks, &task)
+	}
+	err := dumpLogs(tasks, c.Writer)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+}
+
+func dumpLogs(tasks []*TaskModel, w http.ResponseWriter) error {
+	tw := tar.NewWriter(w)
+	defer tw.Close()
+	for _, task := range tasks {
+		err := dumpLog(task.SavedPath, tw)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func dumpLog(savedPath string, tw *tar.Writer) error {
+	f, err := os.Open(savedPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	fi, err := f.Stat()
+	if err != nil {
+		return err
+	}
+	err = tw.WriteHeader(&tar.Header{
+		Name:    path.Base(savedPath),
+		Mode:    int64(fi.Mode()),
+		ModTime: fi.ModTime(),
+		Size:    fi.Size(),
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(tw, f)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// @Summary Retry failed tasks
+// @Description retry tasks that has been failed
+// @Produce json
+// @Success 200 {object} httputil.HTTPSuccess
+// @Failure 500 {object} httputil.HTTPError
+// @Router /logs/tasks/retry [post]
+func (s *Service) TaskRetry(c *gin.Context) {
+	err := s.scheduler.runTaskGroup(true)
```

Comment:
TaskGroupId should be specified in order to correctly work in multi-user env later.